anesthetic/samples.py

```diff
@@ -282,18 +267,15 @@ def plot_2d(self, axes, *args, **kwargs):
             for x, ax in row.iteritems():
                 if (ax is not None and x in self and y in self
                         and ax._upper is not None):
-                    ax_ = ax
                     if x == y:
-                        plot_type = types['diagonal']
-                        kwargs = diagonal_kwargs
-                        ax_ = ax.twin
+                        pos = 'diagonal'
                     elif ax._upper:
-                        plot_type = types['upper']
-                        kwargs = upper_kwargs
+                        pos = 'upper'
                     else:
-                        plot_type = types['lower']
-                        kwargs = lower_kwargs
-                    self.plot(ax_, x, y, plot_type=plot_type, *args, **kwargs)
+                        pos = 'lower'
+                    ax_ = ax.twin if pos == 'diagonal' else ax
```

Comment:
I find `ax_ = ax.twin` shorter than this line. Also one conditional check less.

Reply:
Agreed. One-line conditionals are not very pythonic